starlette/authentication.py

```diff
@@ -23,10 +29,14 @@ def requires(
     scopes: typing.Union[str, typing.Sequence[str]],
     status_code: int = 403,
     redirect: typing.Optional[str] = None,
-) -> typing.Callable[[_CallableType], _CallableType]:
+) -> typing.Callable[
+    [typing.Callable[_P, typing.Any]], typing.Callable[_P, typing.Any]
```

Comment:
Any reason for not using the following?

```suggestion
    [typing.Callable[_P, _T]], typing.Callable[_P, _T]
```

Reply:
Couldn't make it work as it can be async or sync (same issue as https://github.com/encode/starlette/pull/2180#discussion_r1225486748) :/